perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_def_sshPublicKey.java

```diff
@@ -20,17 +28,148 @@
 	@Override
 	public void checkAttributeSyntax(PerunSessionImpl perunSession, User user, Attribute attribute) throws WrongAttributeValueException {
 		//Null in value is ok here
-		if(attribute.getValue() == null) return;
-		
+		if (attribute.getValue() == null) return;
+
 		//Testing if some ssh key contains new line character
 		List<String> sshKeys = attribute.valueAsList();
-		for(String sshKey: sshKeys) {
-			if(sshKey != null) {
-				if(sshKey.contains("\n")) throw new WrongAttributeValueException(attribute, user, "One of keys in attribute contains new line character. New line character is not allowed here.");
+		for (String sshKey : sshKeys) {
+			if (sshKey != null) {
+				if (sshKey.contains("\n"))
+					throw new WrongAttributeValueException(attribute, user, "One of keys in attribute contains new line character. New line character is not allowed here.");
+				try {
+					validateSSH(sshKey);
+				} catch (Exception e) {
+					throw new WrongAttributeValueException("Invalid SSH key format: " + e.getMessage());
+				}
+			}
+		}
+	}
+
+	private void validateSSH(String keyLine) throws Exception {
+		byte[] bytes = null;
+		int[] pos = {0};
+
+		// look for the Base64 encoded part of the line to decode
+		boolean typeFound = false;
+		String typeCheck = "";
+		for (String part : keyLine.split(" ")) {
```

Comment:
If you will check the ssh key like this you can insert random text between the ssh algorithm type and the ssh base64 key. The base64 key has to be put directly after the algorithm type.